impl/forms/ExtractorDerived.h

```diff
@@ -1907,6 +2170,82 @@ class Extractor<Form_C2_sp_store_double> : public Extractor<Form_C2_sp_store>
     Extractor<Form_C2_sp_store_double>(const uint64_t ffmask, const uint64_t fset) :
         Extractor<Form_C2_sp_store>(ffmask, fset)
     {}
+
+    friend class Extractor<Form_C2_sp_store_word_pair>;
+};
+
+/**
+ * Derivative of Form_C2_sp_store_double extractor for STORE WORD PAIR (RV32)
+ */
+template<>
+class Extractor<Form_C2_sp_store_word_pair> : public Extractor<Form_C2_sp_store_double>
+{
+public:
+    Extractor() = default;
+
+    ExtractorIF::PtrType specialCaseClone(const uint64_t ffmask, const uint64_t fset) const override
+    {
+        return ExtractorIF::PtrType(new Extractor<Form_C2_sp_store_double>(ffmask, fset));
+    }
+
+    std::string getName() const override
+    {
+        return Form_C2_sp_store_word_pair::name;
+    }
+
+    uint64_t getSourceRegs(const Opcode icode) const override
+    {
+        // The mask for all of the source regs (addr + data) is
+        // actually the original RS1/RS2 | RS3, but RS3 is not part of
+        // the icode
+        return Extractor<Form_C2_sp_store_double>::getSourceRegs(icode) | getSourceDataRegs(icode);
+    }
+
+    bool isIllop(Opcode icode) const override
+    {
+        // The load pair instruction is illegal if the first operand
+        // is odd
+        const uint32_t reg = extract_(Form_C2_sp_store::idType::RS2, icode);
+        return (reg & 0b1) != 0;
+    }
+
+    uint64_t getSourceDataRegs(const Opcode icode) const override
+    {
+        // Add a second source to the pair
+        uint64_t src_mask = 0;
+        if(const uint32_t reg = extract_(Form_C2_sp_store::idType::RS2, icode); reg != REGISTER_X0)
+        {
+            src_mask = extractUnmaskedIndexBit_(Form_C2_sp_store::idType::RS2, icode, fixed_field_mask_);
+            const uint32_t rs2_val_pos = 64 - __builtin_clzll(src_mask);
+            src_mask |= (0x1ull << rs2_val_pos);
+        }
+        return src_mask;
+    }
+
+    OperandInfo getSourceOperandInfo(Opcode icode, const InstMetaData::PtrType& meta,
+                                     bool suppress_x0 = false) const override
+    {
+        OperandInfo olist;
+        if(const uint32_t reg = extract_(Form_C2_sp_store::idType::RS2, icode); reg != REGISTER_X0)
+        {
+            olist = Extractor<Form_C2_sp_store_double>::getSourceOperandInfo(icode, meta, suppress_x0);
```

Comment:
Doesn't this bypass the addition of the `sp` register?  I think at minimum the register for RS1 needs to be in the list, but not sure.

Reply:
I wasn't quite sure about this. My thinking was that since `Extractor<Form_C2_sp_store_double>` doesn't override `getSourceOperandInfo()`, it will end up calling the base class `Extractor<Form_C2_sp_store>::getSourceOperandInfo()`, which does add RS1 to the source list. 
[https://github.com/sparcians/mavis/blob/b437dce9322e6c6e6e67a4f6d5d5a0a78c93788c/impl/forms/ExtractorForms.h#L556](url)

Should it be changed to just call the base class directly? ie
`olist = Extractor<Form_C2_sp_store>::getSourceOperandInfo(icode, meta, suppress_x0);`

Reply:
With the code as-is, here is the output from running the following directed test:
`./mavis_decode -a rv32 -z -o 0xe00A`

![image](https://github.com/user-attachments/assets/55a9f18d-7b4a-4e9e-b4b0-ec6f7b11bf80)

Reply:
Ok.  That's with `rs2` set to 0?

Reply:
I updated the example to make it more clear that it is actually pulling `sp` into the sources. The previous one I posted was using `rs2 = x2`. Changed it so `rs2 = x4`

Reply:
OK, I believe ya.  :laughing:   Works for me!  Thanks for verifying.